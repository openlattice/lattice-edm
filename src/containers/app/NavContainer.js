/*
 * @flow
 */

import React from 'react';

import styled from 'styled-components';
import { AuthUtils } from 'lattice-auth';
import { NavLink } from 'react-router-dom';

import * as Routes from '../../core/router/Routes';

// injected by Webpack.DefinePlugin
declare var __ENV_DEV__ :boolean;

const NAV_LINK_ACTIVE_CLASSNAME :string = 'nav-link-active';

const Nav = styled.nav`
  background-color: #fefefe;
  border-bottom: 1px solid #c5d5e5;
  display: flex;
  flex: 0 0 auto;
  height: 50px;
  justify-content: center;
  width: 100%;
`;

const NavTab = styled(NavLink).attrs({
  activeClassName: NAV_LINK_ACTIVE_CLASSNAME
})`
  align-items: center;
  border-bottom: 1px solid transparent;
  color: #113355;
  display: flex;
  height: 100%;
  margin: 0 25px;
  text-align: center;
  text-decoration: none;
  &:hover {
   cursor: pointer;
  }
  &.${NAV_LINK_ACTIVE_CLASSNAME} {
    border-bottom: 1px solid #7a52ea;
    color: #7a52ea;
  }
`;

const NavContainer = () => {

  const showSync :boolean = AuthUtils.isAuthenticated() && AuthUtils.isAdmin() && __ENV_DEV__;
  const showGitHub :boolean = AuthUtils.isAuthenticated() && AuthUtils.isAdmin() && __ENV_DEV__;

  return (
    <Nav>
      <NavTab to={Routes.PROPERTY_TYPES}>
        PropertyTypes
      </NavTab>
      <NavTab to={Routes.ENTITY_TYPES}>
        EntityTypes
      </NavTab>
      <NavTab to={Routes.ASSOCIATION_TYPES}>
        AssociationTypes
      </NavTab>
      <NavTab to={Routes.SCHEMAS}>
        Schemas
      </NavTab>
      {
        showSync && (
          <NavTab to={Routes.SYNC}>
            Sync
          </NavTab>
        )
      }
      {
        showGitHub && (
          <NavTab to={Routes.GITHUB}>
            GitHub
          </NavTab>
        )
      }
    </Nav>
  );
};

export default NavContainer;
