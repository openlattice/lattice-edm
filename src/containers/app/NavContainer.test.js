/* eslint-disable no-underscore-dangle */

import 'jest-styled-components';

import React from 'react';
import LatticeAuth from 'lattice-auth';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';

import NavContainer from './NavContainer';

jest.mock('lattice-auth', () => ({
  AuthUtils: {
    isAdmin: jest.fn(),
    isAuthenticated: jest.fn(),
  }
}));

describe('NavContainer', () => {

  afterAll(() => {
    global.__ENV_DEV__ = false;
    global.__ENV_PROD__ = false;
  });

  beforeEach(() => {
    jest.resetModules();
    jest.clearAllMocks();
  });

  describe('snapshots', () => {

    describe('dev', () => {

      beforeAll(() => {
        global.__ENV_DEV__ = true;
        global.__ENV_PROD__ = false;
      });

      test('should match snapshot, admin and authenticated', () => {

        LatticeAuth.AuthUtils.isAdmin = jest.fn().mockImplementation(() => true);
        LatticeAuth.AuthUtils.isAuthenticated = jest.fn().mockImplementation(() => true);

        const nav = shallow(
          <NavContainer />
        );
        expect(toJson(nav)).toMatchSnapshot();
      });

      test('should match snapshot, admin but not authenticated', () => {

        LatticeAuth.AuthUtils.isAdmin = jest.fn().mockImplementation(() => true);
        LatticeAuth.AuthUtils.isAuthenticated = jest.fn().mockImplementation(() => false);

        const nav = shallow(
          <NavContainer />
        );
        expect(toJson(nav)).toMatchSnapshot();
      });

      test('should match snapshot, not admin but authenticated', () => {

        LatticeAuth.AuthUtils.isAdmin = jest.fn().mockImplementation(() => false);
        LatticeAuth.AuthUtils.isAuthenticated = jest.fn().mockImplementation(() => true);

        const nav = shallow(
          <NavContainer />
        );
        expect(toJson(nav)).toMatchSnapshot();
      });

      test('should match snapshot, not admin and not authenticated', () => {

        LatticeAuth.AuthUtils.isAdmin = jest.fn().mockImplementation(() => false);
        LatticeAuth.AuthUtils.isAuthenticated = jest.fn().mockImplementation(() => true);

        const nav = shallow(
          <NavContainer />
        );
        expect(toJson(nav)).toMatchSnapshot();
      });

    });

    describe('prod', () => {

      beforeAll(() => {
        global.__ENV_DEV__ = false;
        global.__ENV_PROD__ = true;
      });

      test('should match snapshot, admin and authenticated', () => {

        LatticeAuth.AuthUtils.isAdmin = jest.fn().mockImplementation(() => true);
        LatticeAuth.AuthUtils.isAuthenticated = jest.fn().mockImplementation(() => true);

        const nav = shallow(
          <NavContainer />
        );
        expect(toJson(nav)).toMatchSnapshot();
      });

      test('should match snapshot, admin but not authenticated', () => {

        LatticeAuth.AuthUtils.isAdmin = jest.fn().mockImplementation(() => true);
        LatticeAuth.AuthUtils.isAuthenticated = jest.fn().mockImplementation(() => false);

        const nav = shallow(
          <NavContainer />
        );
        expect(toJson(nav)).toMatchSnapshot();
      });

      test('should match snapshot, not admin but authenticated', () => {

        LatticeAuth.AuthUtils.isAdmin = jest.fn().mockImplementation(() => false);
        LatticeAuth.AuthUtils.isAuthenticated = jest.fn().mockImplementation(() => true);

        const nav = shallow(
          <NavContainer />
        );
        expect(toJson(nav)).toMatchSnapshot();
      });

      test('should match snapshot, not admin and not authenticated', () => {

        LatticeAuth.AuthUtils.isAdmin = jest.fn().mockImplementation(() => false);
        LatticeAuth.AuthUtils.isAuthenticated = jest.fn().mockImplementation(() => false);

        const nav = shallow(
          <NavContainer />
        );
        expect(toJson(nav)).toMatchSnapshot();
      });
    });

  });

});
