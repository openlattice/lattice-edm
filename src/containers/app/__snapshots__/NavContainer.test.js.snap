// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavContainer snapshots dev should match snapshot, admin and authenticated 1`] = `
<NavContainer__Nav>
  <NavContainer__NavTab
    to="/propertyTypes"
  >
    PropertyTypes
  </NavContainer__NavTab>
  <NavContainer__NavTab
    to="/entityTypes"
  >
    EntityTypes
  </NavContainer__NavTab>
  <NavContainer__NavTab
    to="/associationTypes"
  >
    AssociationTypes
  </NavContainer__NavTab>
  <NavContainer__NavTab
    to="/schemas"
  >
    Schemas
  </NavContainer__NavTab>
  <NavContainer__NavTab
    to="/sync"
  >
    Sync
  </NavContainer__NavTab>
  <NavContainer__NavTab
    to="/github"
  >
    GitHub
  </NavContainer__NavTab>
</NavContainer__Nav>
`;

exports[`NavContainer snapshots dev should match snapshot, admin but not authenticated 1`] = `
<NavContainer__Nav>
  <NavContainer__NavTab
    to="/propertyTypes"
  >
    PropertyTypes
  </NavContainer__NavTab>
  <NavContainer__NavTab
    to="/entityTypes"
  >
    EntityTypes
  </NavContainer__NavTab>
  <NavContainer__NavTab
    to="/associationTypes"
  >
    AssociationTypes
  </NavContainer__NavTab>
  <NavContainer__NavTab
    to="/schemas"
  >
    Schemas
  </NavContainer__NavTab>
</NavContainer__Nav>
`;

exports[`NavContainer snapshots dev should match snapshot, not admin and not authenticated 1`] = `
<NavContainer__Nav>
  <NavContainer__NavTab
    to="/propertyTypes"
  >
    PropertyTypes
  </NavContainer__NavTab>
  <NavContainer__NavTab
    to="/entityTypes"
  >
    EntityTypes
  </NavContainer__NavTab>
  <NavContainer__NavTab
    to="/associationTypes"
  >
    AssociationTypes
  </NavContainer__NavTab>
  <NavContainer__NavTab
    to="/schemas"
  >
    Schemas
  </NavContainer__NavTab>
</NavContainer__Nav>
`;

exports[`NavContainer snapshots dev should match snapshot, not admin but authenticated 1`] = `
<NavContainer__Nav>
  <NavContainer__NavTab
    to="/propertyTypes"
  >
    PropertyTypes
  </NavContainer__NavTab>
  <NavContainer__NavTab
    to="/entityTypes"
  >
    EntityTypes
  </NavContainer__NavTab>
  <NavContainer__NavTab
    to="/associationTypes"
  >
    AssociationTypes
  </NavContainer__NavTab>
  <NavContainer__NavTab
    to="/schemas"
  >
    Schemas
  </NavContainer__NavTab>
</NavContainer__Nav>
`;

exports[`NavContainer snapshots prod should match snapshot, admin and authenticated 1`] = `
<NavContainer__Nav>
  <NavContainer__NavTab
    to="/propertyTypes"
  >
    PropertyTypes
  </NavContainer__NavTab>
  <NavContainer__NavTab
    to="/entityTypes"
  >
    EntityTypes
  </NavContainer__NavTab>
  <NavContainer__NavTab
    to="/associationTypes"
  >
    AssociationTypes
  </NavContainer__NavTab>
  <NavContainer__NavTab
    to="/schemas"
  >
    Schemas
  </NavContainer__NavTab>
  <NavContainer__NavTab
    to="/github"
  >
    GitHub
  </NavContainer__NavTab>
</NavContainer__Nav>
`;

exports[`NavContainer snapshots prod should match snapshot, admin but not authenticated 1`] = `
<NavContainer__Nav>
  <NavContainer__NavTab
    to="/propertyTypes"
  >
    PropertyTypes
  </NavContainer__NavTab>
  <NavContainer__NavTab
    to="/entityTypes"
  >
    EntityTypes
  </NavContainer__NavTab>
  <NavContainer__NavTab
    to="/associationTypes"
  >
    AssociationTypes
  </NavContainer__NavTab>
  <NavContainer__NavTab
    to="/schemas"
  >
    Schemas
  </NavContainer__NavTab>
</NavContainer__Nav>
`;

exports[`NavContainer snapshots prod should match snapshot, not admin and not authenticated 1`] = `
<NavContainer__Nav>
  <NavContainer__NavTab
    to="/propertyTypes"
  >
    PropertyTypes
  </NavContainer__NavTab>
  <NavContainer__NavTab
    to="/entityTypes"
  >
    EntityTypes
  </NavContainer__NavTab>
  <NavContainer__NavTab
    to="/associationTypes"
  >
    AssociationTypes
  </NavContainer__NavTab>
  <NavContainer__NavTab
    to="/schemas"
  >
    Schemas
  </NavContainer__NavTab>
</NavContainer__Nav>
`;

exports[`NavContainer snapshots prod should match snapshot, not admin but authenticated 1`] = `
<NavContainer__Nav>
  <NavContainer__NavTab
    to="/propertyTypes"
  >
    PropertyTypes
  </NavContainer__NavTab>
  <NavContainer__NavTab
    to="/entityTypes"
  >
    EntityTypes
  </NavContainer__NavTab>
  <NavContainer__NavTab
    to="/associationTypes"
  >
    AssociationTypes
  </NavContainer__NavTab>
  <NavContainer__NavTab
    to="/schemas"
  >
    Schemas
  </NavContainer__NavTab>
</NavContainer__Nav>
`;
